src/gateways/createBookConfigLoader.ts

```diff
@@ -0,0 +1,46 @@
+import { BOOKS } from '../config';
+import { acceptStatus } from '../helpers/fetch';
+import Sentry from '../helpers/Sentry';
+
+export interface BookVersion {
+  defaultVersion: string;
+}
+
+interface BookConfig {
+  [key: string]: BookVersion;
+}
+
+interface ReleaseJsonStructure {
+  books: BookConfig;
+  code: string;
+  id: string;
+}
+
+const cachedBooks = { ...BOOKS } as BookConfig;
+
+export default (baseUrl: string) => {
+  const loadRemoteBookConfig = (): Promise<BookConfig | undefined> => {
+    const url = `${baseUrl}/rex/release.json`;
+    return fetch(url)
+      .then(acceptStatus(200, (status, message) => `Error response from "${url}" ${status}: ${message}`))
+      .then((response) => response.json() as Promise<ReleaseJsonStructure>)
+      .then((response) => response && response.books)
+      .catch((e) => {
+        Sentry.captureException(e);
+        return Promise.resolve(undefined);
+      });
+  };
+
+  return {
+    getBookVersionFromUUID: (uuid: string): BookVersion | Promise<BookVersion | undefined> => {
```

Comment:
my understanding was that we should have 2 methods - one synchronous and the second one asynchronous 
both of them should access `cachedBooks` variable
the async one will be used in `resolveContent` and it will update `cachedBooks` if it wasn't found previously
the sync one will just return version from `cachedBooks`

Reply:
To be consistent I will have to replace all `BOOKS[id]` calls in the project with the sync function from the `bookConfigLoader`, right? But in the other hand it will make some components need to receive or load `AppServices['BookConfigLoader']`, which will require some refactoring (like in this [PR](https://github.com/openstax/rex-web/pull/1153), but without making some functions async).

Reply:
I think this will be okay. The other option would be to just create a module that exports these functions instead of creating it as a gateway and do not pass it to the services

Reply:
@TomWoodward Do you have an opinion on this?

Reply:
it might be ok for the synchronous function to be a regular export that references a `let books` at the top of the module